manifests/init.pp

```diff
@@ -18,14 +23,28 @@
     }
   }
 
-  file { $package_location:
-    ensure  => file,
-    source  => $source,
-    mode    => '0644',
-    backup  => false,
-    require => File["${facts['puppet_vardir']}/packages"],
+  if $remote_package {
```

Comment:
instead of having the `$remote_package` parameter, what if you just looked at the source and used remote file if it wasn't `puppet:///*`?

Reply:
So something like `if 'puppet:///' not in $source`? That gives you the edge case of someone having a stupid url like that on their webserver, but my friday brain doesn't want to do the regex.